gprofiler/profilers/java.py

```diff
@@ -1025,6 +1026,9 @@ def _select_processes_to_profile(self) -> List[Process]:
             # profiling.
         return pgrep_maps(DETECTED_JAVA_PROCESSES_REGEX)
 
+    def _should_profile_process(self, process: Process) -> bool:
+        return any(line.endswith("/libjvm.so") for line in read_proc_file(process, "maps").splitlines())
```

Comment:
Use `DETECTED_JAVA_PROCESSES_REGEX` (and just `re.search()` on the file can be enough, no need for line split)